File: infrastructure/db-core/src/main/java/com/nmh/commerce/coupon/IssuedCouponRepositoryImpl.java
package com.nmh.commerce.coupon;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@RequiredArgsConstructor
@Repository
public class IssuedCouponRepositoryImpl implements IssuedCouponRepository {
    private final IssuedCouponJpaRepository issuedCouponJpaRepository;


    @Override
    public IssuedCoupon save(IssuedCoupon issuedCoupon) {
        return issuedCouponJpaRepository.save(IssuedCouponEntity.from(issuedCoupon)).toDomain();
    }
}

File: infrastructure/db-core/src/main/kotlin/com/nmh/commerce/coupon/IssuedCouponRepositoryImpl.kt
package com.nmh.commerce.coupon

import org.springframework.stereotype.Repository

@Repository
class IssuedCouponRepositoryImpl(private val issuedCouponJpaRepository: IssuedCouponJpaRepository) : IssuedCouponRepository {
    override fun save(issuedCoupon: IssuedCoupon): IssuedCoupon {
        return issuedCouponJpaRepository.save(IssuedCouponEntity.from(issuedCoupon)).toDomain()
    }
}
